src/parquet/encodings/dictionary-encoding.h

```diff
@@ -64,22 +64,17 @@ class DictionaryDecoder : public Decoder<Type> {
 
   virtual int Decode(T* buffer, int max_values) {
     max_values = std::min(max_values, num_values_);
-    for (int i = 0; i < max_values; ++i) {
-      buffer[i] = dictionary_[index()];
+    int decoded_values = idx_decoder_.GetBatchWithDict(buffer, max_values, dictionary_);
+    if (decoded_values != max_values) {
+      ParquetException::EofException();
     }
```

Comment:
Can we remove the exception here and subtract, return `decoded_values` below ?
The semantics of `Decode` like methods should not worry about the mismatch in the values. The layers using these methods should throw an exception if required.